packages/apps/reputation-oracle/server/src/modules/auth/auth.service.ts

```diff
@@ -265,6 +277,51 @@ export class AuthService {
     return this.generateTokens(userEntity.id, jwtPayload);
   }
 
+  private async checkStakeEligible(
+    userEntity: Web2UserEntity | UserEntity,
+  ): Promise<boolean> {
+    if (!this.stakingConfigService.eligibilityEnabled) return true;
+
+    const apiKeys = await this.exchangeApiKeysService.retrieve(userEntity.id);
+
+    let inspectedStakeAmount = 0;
+
+    if (apiKeys) {
+      try {
+        const client = await this.exchangeClientFactory.create(
+          apiKeys.exchangeName as SupportedExchange,
+          {
+            apiKey: apiKeys.apiKey,
+            secretKey: apiKeys.secretKey,
+          },
+          { timeoutMs: this.stakingConfigService.timeoutMs },
+        );
+        const exchangeBalance = await client.getAccountBalance(
+          this.stakingConfigService.asset,
+        );
+        inspectedStakeAmount += exchangeBalance;
+      } catch (err) {
+        this.logger.warn('Failed to query exchange balance; continuing', {
+          userId: userEntity.id,
+          exchangeName: apiKeys.exchangeName,
+          error: err,
+        });
+      }
+    }
+
+    if (
+      inspectedStakeAmount < this.stakingConfigService.minThreshold &&
+      userEntity.evmAddress
+    ) {
+      const onChainStake = await this.web3Service.getStakedBalance(
+        userEntity.evmAddress,
+      );
+      inspectedStakeAmount += onChainStake;
+    }
+
+    return inspectedStakeAmount >= this.stakingConfigService.minThreshold;
```

Comment:
@flopez7 
If there is a network error/hiccup, then `inspectedStakeAmount` might be 0 and user on UI will see that they don't have correct stake.

Would it be better to fail auth in this case? Or maybe have some other way to handle that taking into account that info is written in JWT and also used by oracles